ipns/ipns.go

```diff
@@ -169,3 +202,23 @@ func (m *Manager) getSemaphore(key string) chan struct{} {
 	}
 	return ptl
 }
+
+func (m *Manager) republish() error {
+	ctx, cancel := context.WithTimeout(context.Background(), listKeysTimeout)
+	defer cancel()
+	start := time.Now()
+	keys, err := m.keys.List(ctx)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+	withPath := 0
+	for _, key := range keys {
```

Comment:
Doing these one at a time will likely take hours, maybe more than 4 hours, which may be the cause of our repub pain in general (assuming IPFS is doing something similar internally). We can at least do them in batches 50 or something (just picked that number). Here's an example of this kind of parallel loop processing in batches: https://github.com/textileio/textile/blob/edbdd05058fccc40e8c23817bf4e44288e59208a/buckets/local/pull.go#L146 (down to line 164).

Reply:
just wrap this guy in a batch like your link, or are you thinking i should batch up something lower like this step https://github.com/textileio/textile/pull/493/files#diff-565b03391ddf3dcb3b46674b1eec6d37ca610e29cefb834690fab96955ca89fcR148

Reply:
Yeah just batch the calls to `m.publish`

Reply:
cool, done